### PR TITLE
13 get api

### PR DESCRIPTION
**Pull request by kalumwildig**

### CODE REVIEW

File: models/model.js [hunk 1]
```diff
@@ -75,3 +74,6 @@ exports.postCommentModel = async (data, id, author) => {
     return Promise.reject((err.code = "22P02"));
```
Comment (douglashellowell) [inline models/model.js:74]: You are invoking `Promise.reject` with some code that is assinging a value to a key in an object... that doesn't exist! 😱

You can invoke Promise.reject with the reason. The way we have shown it in lectures is passing a status code and message

```js
Promise.reject( { status: 400, msg: "invalid id" } )
```

Have a look at how you are checking errors in your error handler - your `if` statement is using as assignment operator - not a comparason operator!

**Review by douglashellowell — CHANGES_REQUESTED** [review]
Code from this PR is good but I've spotted some mistakes in how you are using `Promise.reject` and an `if`statement - please change these!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -387,15 +387,27 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+});
+
+describe("GET /api", () => {
+  test("Status 200: Should return JSON of all the potential endpoints", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body: { endpoints } }) => {
+        expect(typeof endpoints).toBe('object')
+        expect(Object.keys(endpoints)).toHaveLength(8)
```
Comment (popatre) [inline __tests__/app.test.js:410]: Only problem with this, is when you add more endpoints, and update that file, you'll have to keep updating this test. 
I'd just expect it to be an object, OR keep updating this.

File: endpoints.json [hunk 3]
```diff
@@ -11,17 +11,79 @@
...
+        "created_at": "2020-07-09T20:11:00.000Z",
+        "votes": 88
+      }
+    }
+  },
+  "POST /api/articles/:article_id/comments": {
+    "description": "inserts a comment for a given article id and serves a string of the added comment",
+    "queries": [],
+    "exampleResponse": {
+      "comment": "This is an example response."
```
Comment (popatre) [inline endpoints.json:86]: Does this not respond with the whole comment object? i.e. votes, author etc?

File: endpoints.json [hunk 4]
```diff
@@ -11,17 +11,79 @@
...
+        "topic": "mitch",
+        "author": "butter_bridge",
+        "body": "I find this existence challenging",
+        "created_at": "2020-07-09T20:11:00.000Z",
+        "votes": 88
+      }
+    }
+  },
+  "POST /api/articles/:article_id/comments": {
+    "description": "inserts a comment for a given article id and serves a string of the added comment",
```
Comment (popatre) [inline endpoints.json:83]: - [ ] Add an exampleRequest - what does the post body have to look like?

**Review by popatre — CHANGES_REQUESTED** [review]

File: endpoints.json (hunk 3 above)
Comment (antariess) [reply endpoints.json:88]: i believe this feedback is still outstanding

**Pull request merged**